doctr/utils/geometry.py

```diff
@@ -230,9 +230,13 @@ def estimate_page_angle(polys: np.ndarray) -> float:
     """Takes a batch of rotated previously ORIENTED polys (N, 4, 2) (rectified by the classifier) and return the
     estimated angle ccw in degrees
     """
+    # Compute mean left points and mean right point with respect to the reading direction (oriented polygon)
+    xleft = polys[:, 0, 0] + polys[:, 3, 0]
+    yleft = polys[:, 0, 1] + polys[:, 3, 1]
+    xright = polys[:, 1, 0] + polys[:, 2, 0]
+    yright = polys[:, 1, 1] + polys[:, 2, 1]
     return np.median(np.arctan(
-        (polys[:, 0, 1] - polys[:, 1, 1]) /  # Y axis from top to bottom!
-        (polys[:, 1, 0] - polys[:, 0, 0])
+        (yleft - yright) / (xright - xleft)  # Y axis from top to bottom!
```

Comment:
Now this looks quite logical, but I think we're missing a unittest somewhere. This will change the existing behaviour for sure, but for some reasons, the unittests are still passing. Could you check please? :pray:

Reply:
I added a test!

Reply:
But this doesn't really modify the behaviour of this function, it is just more accurate/robust to skewed cases so I guess it is expected that we don't break unitests here

Reply:
I meant: that change will change the output value for a given input if the polygon is not a straight bounding box